ci/travis_before_script_cpp.sh

```diff
@@ -53,7 +53,8 @@ CMAKE_COMMON_FLAGS="\
 -DARROW_NO_DEPRECATED_API=ON \
 -DARROW_EXTRA_ERROR_CONTEXT=ON"
 CMAKE_LINUX_FLAGS=""
-CMAKE_OSX_FLAGS=""
+CMAKE_OSX_FLAGS="\
+-DARROW_GFLAGS_USE_SHARED=OFF"
```

Comment:
Because `@rpath` can't be resolved with GFlags installed by conda-forge.

https://travis-ci.org/kou/arrow/jobs/494364820#L1993

> ```text
>  2/111 Test   #1: plasma-serialization_tests ...................***Failed    1.36 sec
>Running plasma-serialization_tests, redirecting output into /Users/travis/build/kou/arrow/cpp-build/build/test-logs/plasma-serialization_tests.txt (attempt 1/1)
>dyld: Library not loaded: @rpath/libgflags.2.2.dylib
>  Referenced from: /Users/travis/build/kou/arrow/cpp-install/lib/libarrow.13.dylib
>  Reason: image not found
> ```

We will be able to resolve it by running each test with `DYLD_LIBRARY_PATH` environment variable:

```shell
DYLD_LIBRARY_PATH=${MINICONDA}/lib ${ARROW_BUILD_TYPE}/plasma-serialization_tests
```

Because `DYLD_LIBRARY_PATH` can affect only the child process.

It's not useful. So I use static library than shared library.

Reply:
This is working for all other libraries? Then there must be something wrong with `libgflags`.

Reply:
Other libraries are used as static library. So they work.

`gflags-config.cmake` provides `@rpath/libgflags.2.2.dylib` for `IMPORTED_LOCATION`. If `gflags-config.cmake` doesn't use `@rpath`, this will not be caused.

Generally, using `@rpath` isn't bad. GFlags installed from conda-forge can't assume where it's installed. So I think that using `@rpath` is reasonable for this case.

We will be able to resolve `@rpath` in our `FindGFlags.cmake`. This is another solution.

We can specify `DYLD_LIBRARY_PATH` to resolve `@rpath` in `cpp/build-support/run-test.sh`. This is also another solution.